### DISCUSSION TITLE
Feature: adding a data broker to the application

### DISCUSSION THREAD

**Discussion by hectron** [happy-software/sleuth#2] [Ideas]
The purpose behind this application is to opt-out of data brokers that might have acquired a person's information after the fact. I'm using this discussion as a way to figure out how to go about solving this problem.

# First pass

I think initially, we should do a simple script/rake task for a single data aggregator (my suggestion is to start with `spokeo`, one of the most popular data aggregators)

- search for a specific person
- confirm that specific person is the user of this application
- try to automate opt-ing out
  - if unable to automatically opt-out, provide a link to the opt-out page and ask the user if they've been able to opt out
- once opted-out, mark that user as opted out of the aggregator (with a timestamp of when)

This approach should lay out the foundation for how to application should work.

**Comment by hebron-george** [comment]
Maybe a feature to scrape email for unsubscribe links and list them out all together for the user to quickly unsub from a bunch?

**Reply by hectron** [reply]
I like this idea a lot! It's a little different from the original `data aggregator` functionality, but I think it relates to the idea of opting out of communications from companies/services/people.

I'll make this into a separate discussion, because I think that it has potential.

**Comment by hebron-george** [comment]
> The purpose behind this application is to opt-out of data aggregators that might have acquired a person's information after the fact. I'm using this discussion as a way to figure out how to go about solving this problem.
> 
> # First pass
> I think initially, we should do a simple script/rake task for a single data aggregator (my suggestion is to start with `spokeo`, one of the most popular data aggregators)
> 
> * search for a specific person
> * confirm that specific person is the user of this application
> * try to automate opt-ing out
>   
>   * if unable to automatically opt-out, provide a link to the opt-out page and ask the user if they've been able to opt out
> * once opted-out, mark that user as opted out of the aggregator (with a timestamp of when)
> 
> This approach should lay out the foundation for how to application should work.

Aren't these data aggregators paid services?

Also it might be easier to go the other way instead - i.e. find a bunch of services that are easy to opt out of and we manually aggregate that list. Then we give a user an option to input their email/identifier to opt out from all the easy ones at once?

**Reply by hectron** [reply]
So -- I should have updated this but I meant to say a Data Broker. For example, Spokeo has information about most people but a user can opt out by requesting their profile be removed.

I think the way that you're suggesting makes sense. [I found this list of data brokers](https://github.com/yaelwrites/Big-Ass-Data-Broker-Opt-Out-List), and was thinking of placing them in code somehow. But I wanted to focus on a single one at first.
